fdbserver/workloads/ConsistencyCheck.actor.cpp

```diff
@@ -1946,20 +1996,82 @@ struct ConsistencyCheckWorkload : TestWorkload {
+			state KeyRangeMap<bool>::Ranges failedRangesList = failedRanges.ranges();
+			state KeyRangeMap<bool>::iterator failedRangesIter = failedRangesList.begin();
+			for (; failedRangesIter != failedRangesList.end(); ++failedRangesIter) {
+				if (failedRangesIter->value() == false) {
+					continue; // No failure, so bypass the range
+				}
+				state KeyRange failedRange = failedRangesIter->range();
+				state Promise<std::vector<std::pair<KeyRange, std::vector<StorageServerInterface>>>> keyServerPromise;
+				bool keyServerResult =
+				    wait(self->getKeyServers(cx, self, keyServerPromise, failedRange.withPrefix(keyServersPrefix)));
+				state bool missingFailedRange = true;
+				if (keyServerResult) {
+					state std::vector<std::pair<KeyRange, std::vector<StorageServerInterface>>> keyServers =
+					    keyServerPromise.getFuture().get();
+					// Get the locations of all the shards in the database
+					state Promise<Standalone<VectorRef<KeyValueRef>>> keyLocationPromise;
+					bool keyLocationResult = wait(self->getKeyLocations(cx, keyServers, self, keyLocationPromise));
+					if (keyLocationResult) {
+						Standalone<VectorRef<KeyValueRef>> keyLocationsForFailedRanges =
+						    keyLocationPromise.getFuture().get();
+						for (int k = 0; k < keyLocationsForFailedRanges.size() - 1; k++) {
+							shardLocationPairListForFailedRanges.push_back(
+							    std::make_pair(Standalone(KeyRangeRef(keyLocationsForFailedRanges[k].key,
+							                                          keyLocationsForFailedRanges[k + 1].key)),
+							                   Standalone(keyLocationsForFailedRanges[k].value)));
+						}
+						missingFailedRange = false;
+					}
+				}
+				if (missingFailedRange) {
+					TraceEvent(SevWarnAlways, "ConsistencyCheck_MissingRange")
+					    .setMaxEventLength(-1)
+					    .setMaxFieldLength(-1)
+					    .detail("Reason", "Failed to do getKeyServers or getKeyLocations")
+					    .detail("ConsistencyCheckEpoch", consistencyCheckEpoch)
+					    .detail("Distributed", self->distributed)
+					    .detail("ClientId", self->clientId)
+					    .detail("ClientCount", self->clientCount)
+					    .detail("BeginKey", printable(failedRange.begin))
+					    .detail("EndKey", printable(failedRange.end));
+				}
+			}
+
+			if (shardLocationPairListForFailedRanges.size() > 0) {
+				if (consistencyCheckEpoch < CLIENT_KNOBS->CONSISTENCY_CHECK_RETRY_DEPTH_MAX) {
+					// Check that each failed shard has the same data on all storage servers that it resides on
+					wait(::success(self->checkDataConsistency(cx,
+					                                          shardLocationPairListForFailedRanges,
+					                                          configuration,
+					                                          tssMapping,
+					                                          self,
+					                                          ++consistencyCheckEpoch)));
```

Comment:
This changes `consistencyCheckEpoch` in this actor. Maybe pass `consistencyCheckEpoch + 1` instead?

Reply:
Yes. This is more clear and portable.